metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -299,58 +363,71 @@ def _get_upstream_lineage_info(
+            with PerfTimer() as timer:
+                self._populate_view_downstream_lineage(conn)
+                self.report.view_downstream_lineage_query_secs = timer.elapsed_seconds()
 
     def _populate_external_lineage(self, conn: SnowflakeConnection) -> None:
-        # Handles the case where a table is populated from an external location via copy.
-        # Eg: copy into category_english from 's3://acryl-snow-demo-olist/olist_raw_data/category_english'credentials=(aws_key_id='...' aws_secret_key='...')  pattern='.*.csv';
-        query: str = SnowflakeQuery.external_table_lineage_history(
-            start_time_millis=int(self.config.start_time.timestamp() * 1000)
-            if not self.config.ignore_start_time_lineage
-            else 0,
-            end_time_millis=int(self.config.end_time.timestamp() * 1000),
-        )
 
         num_edges: int = 0
         self._external_lineage_map = defaultdict(set)
-        try:
-            for db_row in self.query(conn, query):
-                # key is the down-stream table name
-                key: str = self.get_dataset_identifier_from_qualified_name(
-                    db_row["DOWNSTREAM_TABLE_NAME"]
-                )
-                if not self._is_dataset_pattern_allowed(key, "table"):
-                    continue
-                self._external_lineage_map[key] |= {
-                    *json.loads(db_row["UPSTREAM_LOCATIONS"])
-                }
-                logger.debug(
-                    f"ExternalLineage[Table(Down)={key}]:External(Up)={self._external_lineage_map[key]} via access_history"
-                )
-        except Exception as e:
-            self.warn(
-                "external_lineage",
-                f"Populating table external lineage from Snowflake failed."
-                f"Please check your premissions. Continuing...\nError was {e}.",
+        if self.report.edition == SnowflakeEdition.STANDARD:
+            logger.info(
+                "Snowflake Account is Standard Edition. External Lineage Feature via Access History is not supported."
             )
+        else:
+            # Handles the case where a table is populated from an external location via copy.
+            # Eg: copy into category_english from 's3://acryl-snow-demo-olist/olist_raw_data/category_english'credentials=(aws_key_id='...' aws_secret_key='...')  pattern='.*.csv';
+            query: str = SnowflakeQuery.external_table_lineage_history(
+                start_time_millis=int(self.config.start_time.timestamp() * 1000)
+                if not self.config.ignore_start_time_lineage
+                else 0,
+                end_time_millis=int(self.config.end_time.timestamp() * 1000),
+            )
+
+            try:
+                for db_row in self.query(conn, query):
+                    # key is the down-stream table name
+                    key: str = self.get_dataset_identifier_from_qualified_name(
+                        db_row["DOWNSTREAM_TABLE_NAME"]
```

Comment:
Is accessing dictionaries like this safe? I'd recommend we don't do this anywhere. Even if the documentation says this will be populated, we need a safer way to access.

Reply:
In this specific case it's fine, since it's coming from the result of a SQL query and hence will be populated